### PR TITLE
[RetinaNet Refactor] Fix bug and Reuse box delta encoding and decoding logic.

### PR DESCRIPTION
**Pull request by tanzhenyu**
Fixing bug -- The original implementation uses "XYWH" for encoding and decoding. The paper uses "Center_XYWH" for encoding and decoding
The result is ~34.0mAP at 25 epochs.

### CODE REVIEW

File: keras_cv/bounding_box/converters.py [hunk 1]
```diff
@@ -30,11 +31,16 @@ def _encode_box_to_deltas(
     boxes: tf.Tensor,
     anchor_format: str,
     box_format: str,
-    variance: Optional[List[float]] = None,
+    variance: Optional[Union[List[float], tf.Tensor]] = None,
```
Comment (LukeWood) [inline keras_cv/bounding_box/converters.py:34]: is this type annotation useful still or does it become overly burdensome?  No strong feelings, just might be worth dropping it

**Review by LukeWood — APPROVED** [review]
LGTM overall, nice that it boosts the MaP a bit!

File: keras_cv/bounding_box/converters.py (hunk 1 above)
Comment (tanzhenyu) [reply keras_cv/bounding_box/converters.py:34]: To me, it feels burdensome. But whenever I read someone else's code with type annotation, I still feel it's pretty useful. So it's really a double-edged sword IMHO

**Review by ianstenbit — APPROVED** [review]
Thanks!

**Pull request merged**